Article first draft (#135)
hassanobeid1994/tr_b_causal_2020: article/sections/_7_latent_confounding.tex

> \section{Latent Confounding}

Author: cc @hassan-obeid: Somewhere in this section, perhaps in a footnote, can you link to the project repository and where you perform the various simulations and investigations (i.e., your public notebooks)?

\section{Latent Confounding}
\label{sec:latent-confounding}

> This section focuses on the more complicated yet realistic case that is 
> typically faced by demand modelers, where we have latent confounders that 
> affect the treatment assignment of our causal variables of interest, as well as the outcome. 

Author: I think this sentence is doing 3 things. It states the focus of this section; it describes the focus of the section; and it defines latent confounders. I think all three of these things are important, and I think they should all happen in the first paragraph. I suggest splitting these three functions into three sentences to hopefully achieve the highest reader comprehension.
```suggestion
This section focuses on latent confounding in causal inference problems.
As mentioned in Section \ref{sec:graph-construction},  latent confounders affect the treatment assignment of our causal variables of interest, as well as the outcome.
While more complicated, this setup is the more realistic and typical case faced by demand modelers.
```

We first go over a few examples of confounding in transportation analyses, explain 

> the challenges that come with such cases, and present a few approaches to 

Author: What approaches are presented besides the deconfounder? I think this may be an unfulfilled promise. However, there SHOULD be a subsection that provides reference to the array of other approaches that can/should be investigated for one's latent confounding problems.

dealing with this issue, specifically the recent de-confounder technique of 
\citet{wang_2019_blessings}. 
Our application will show how directed acyclic graphs can 

> help increase transparency about one's reasoning regarding the number of 

Author: ```suggestion
help clarify one's reasoning regarding the number of 
```

> confounders, assumptions for which variables are confounded, and the models 

Author: For standardization across the list: "clarify one's reasoning; clarify one's assumptions; clarify the models"
```suggestion
confounders, one's assumptions for which variables are confounded, and the models 
```

needed to estimate the causal effects. 

> In addition to the application above, we use simplified simulation scenarios to further investigate the usefulness 

Author: Do you write about both an application and the simulation? On first glance, I see only the section "Case study: Simulation." If there is an analogous collection of paragraphs about an application, we should add a label for it!

and pitfalls of this approach for generating accurate model estimates. 

\subsection{Latent Variable Models}
\label{intro-latent-models}

> % TODO: Mechanics of how latent variable models control for unobserved confounding.

Author: I thought the transition from the last paragraph was pretty jarring. Can you add a sentence that provides a bridge from latent confounding to latent variables and latent variable models? And yes I separated those two on purpose---your section is about latent variable models, but your current first sentence begins talking about latent variables immediately.

Reviewer: @bouzaghrane can you update this paragraph/section accordingly?

Latent variables have been used time and again in different applications of statistical and econometric models in different fields, ranging from epidemiology to transportation.
A latent variable is defined as a covariate not directly observed about subjects in the sample.
Examples of these variables include measurements about happiness, lifestyle, economic expectation, 

> morale, among others in many different fields.

Author: All of these examples, while true, are inherently unobservable. I think it's important to **also** show that latent variable models aren't only for psycological constructs (attitudes, subjective expectations, etc.). I think this will help the largest amount of people find relevance in our work.

Some examples of latent variables that are not inherently unobservable include the following:
- measurement error (true measurement is latent, we get noisy proxy or nothing)
   - medical laboratory measurements with human/mechanical error or impreciseness (e.g. blood pressure)
   - travel time for chosen modes (was it actually 30 min or was it 33 min?), 
   - income category instead of income
- survey item non-response (e.g. somebody simply doesn't respond to a survey question)
- missing data (e.g. tech companies that have servers go down so they are truly and simply missing the data)

Reviewer: @bouzaghrane

Latent variable models are based on the assumption that observed variables are independent conditional on latent variables.

> Examples of these models include Integrated Choice and Latent Variable Models, Generalized Linear Latent and Mixed Models,
> Factor Analysis Models, and Latent Class Models.

Author: I know this is just semantics, but I suggest adding hierarchical models to this list. It's semantics because all latent variable models are hierarchical. However, I think we should explicitly call out this connection as it provides another avenue for a reader to understand and connect with our work. Someone may be familiar with the term hierarchical model and then better able to understand our points because we connected to them!

For the same reason I would parenthetically call out mixed logit models. Finally, I changed to the ICLV acronym because it's defined in Section 3.
```suggestion
Examples of these models include ICLV, Generalized Linear Latent and Mixed Models (GLMM; including mixed logit models),
Factor Analysis Models, and Latent Class Models.
```

Author: Also, now that you've mentioned latent variable models, can you explicitly relate the latent variable models to latent confounding? For instance, can you reference any previous studies or literature that uses or recommends latent variable models to address latent confounding?

Reviewer: @bouzaghrane

> Latent variable models usually rely on collecting indicator/proxy data to infer the levels of latent variable for individuals in a certain sample.

Author: I think this may be incorrect. Out of your list, only ICLV models rely on collecting indicator/proxy data. GLMM/MixedLogit, Latent Class, and Factor Analysis models don't rely on such data to infer their latent variables. And on top of this, I think ICLVs are the least used of all these types of models. I haven't even really seen ICLVs outside of transportation and econometrics. The other models are all more cross-disciplinary, in my opinion. Factor analysis is used everywhere (aka every recommendation system). GLMMs are tools for fancy statisticians regardless of discipline. And finite mixture models (aka super simple latent class models) have historically long use everywhere from computer science to operations research...

Reviewer: @bouzaghrane

> The collection of this data is not always easy or even feasible, mainly due to potential sensitivity of subjects towards some attitudinal questions

Author: ```suggestion
The collection of this data is not always easy or even feasible, mainly due to potential sensitivity of subjects towards some attitudinal questions.
```

> This makes it difficult for researchers to collect necessary information that allows for controlling for unobserved factors in models. 

Author: In general, I wonder about the purpose of this paragraph. It seems the purpose is to highlight that life is hard without indicator variables. This is true, but again I wonder about your larger point. I think you're getting at the notion that "it's hard to build a good latent variable model without indicator/proxy data, therefore one should expect lesser performance from approaches that don't rely on such indicators." If I'm correct, then I think it'd be good to state this directly at the end of the paragraph.

Reviewer: @bouzaghrane agree

\subsection{Examples of confounding}
\label{sec:confounding-examples}

Confounding occurs when a certain (confounding) variable induces variations in
both the outcome as well as the treatment (policy) variables of interest, 
creating correlation between the treatment and the outcome that is not caused

> by the treatment variable itself. When the confounding variable (or variables) is 

Author: Why do you define confounding variables again when you've already done so in the first sentence of the section? I'm not sure that its super important to highlight the fact that confounders induce correlation between treatment and (potential) outcomes.

Author: Ah, I see, you return to concepts from this sentence in the last sentence of the paragraph. Alright. Could you appease the attentive reader with something like the following though (so they know the repetition is purposeful):
```suggestion
We've already noted that confounding occurs when a certain (confounding) variable induces variations in
both the outcome as well as the treatment (policy) variables of interest. 
These dual effects create a correlation between the treatment and the outcome that is not caused
by the treatment variable itself. When the confounding variable (or variables) is 
```

methods in the literature for how to do that, including post-stratification,
multiple regression, propensity score methods, etc. It is when the 
confounding variable is unobserved that the problem becomes significantly 
more challenging. 
For an illustration, imagine we're interested in the effect of adding a bike on the mode share of bicycling in a given neighborhood. 
To estimate this effect, one strategy that a modeler may use is to develop a disaggregate mode choice model with a dummy variable for whether a bikelane exists between an individual's trip origin and destination, and then treating the coefficient on this variable as the causal effect of adding a bikelane on the log-odds of choosing to bike. 
The problem with such strategy is that individuals may self-select to live in an area where bicycle infrastructure exists because they have a preference for commuting by bike. 
In other words, if we define an additional variable to encode a person's latent inherent preference for biking, then this variable determines both a person's likelihood for living in an area with an existing bicycle infrastructure, as well as whether that person choose to bike. 

> This variable is subsequently expected not to be "balanced" across the treatment and control groups; those who live near a bicycle lane are expected to have higher preference for biking than those who don't. 

Author: ```suggestion
This variable is subsequently expected to lack ``balance'' across the treatment and control groups; those who live near a bicycle lane are expected to have higher preference for biking than those who don't. 
```

Without accounting for this latent confounder, we risk getting very biased estimates 
of the effects of our treatment variable of interest, created by the 
induced variations by the confounder in both the treatment and the outcome 
variable. 

> The problem of latent confounding and ommitted variable bias is widely 
> acknowledged in the transportation literature, and demand modelers can draw on 
> a list of method to account for confouding in some specific circumstances. 

Author: Hmm, references should definitely be provided here. Omitted variable bias is widely acknowledged, but not necessarily (or explicitly) omitted variable bias caused by latent confounding. Either way we should provide citations for these biases being acknowledged. We should also provide a list of citations for methods to account for confounding. This will be helpful all by itself!
```suggestion
The problem of latent confounding and ommitted variable bias is widely 
acknowledged in the transportation literature, and demand modelers can draw on 
a list of methods to account for confounding in limited circumstances.
```

> Integrated choice and latent variable (ICLV) models are a way to account for 
> the effect of unobserved attitudinal variables which may affect the selection 
> of some individuals into some treatment level, as well as an outcome of 
> interest. For example, a person's unobserved beliefs and attitudes towards 

Author: Beyond the first sentence, this paragraph repeats a lot of information from above. In particular, the first paragraph in the "latent variable models" section describes attitudinal confounders without naming them as such. ICLV models are also introduced there as a method for dealing with such confounding. The last paragraph above also featured an example of an attitudinal confounder in the context of bicycling. Similarly, the last paragraph before the section header "Examples of confounding" also describes difficulties with collecting indicator variables.

In general, I wonder if this paragraph (lines 61-71) can be removed and the relevant parts can be added elsewhere in the section above?
```suggestion
Integrated choice and latent variable (ICLV) models are a way to account for 
the effect of latent, attitudinal confounders. For example, a person's unobserved beliefs and attitudes towards 
```

> being environmentelly friendly may both affect whether she chooses to bike to 

Author: ```suggestion
being environmentally friendly may both affect whether she chooses to bike to 
```

> work, as well as whether she lives close to a bike infrastructure, which may 

Author: I'm suggesting a sentence break so your sentence doesn't attempt to express many ideas at once.
```suggestion
work, as well as whether she lives close to a bike infrastructure. These unobserved relations may 
```

bias our observational study if we're interested in the effect adding a 
bikelane has on the mode share of bicycles. Those methods, however, rely on 
collecting attitudinal indicators typically obtained by conducting
additional surveys with a sample of the people. This may not be an option in 
many cases; for example, we may not be able to reach the individuals to conduct additional surveys with them. 

\subsection{The deconfounder algorithm}
\label{sec:deconfounder-algo}

One recent method that has been proposed to deal with the problem of latent 
confounding without the need to collect additional data is the deconfounder 
algorithm by \citet{wang_2019_blessings}. The method attempts to control for the 

> confouding variable by estimating a "subsitute confounder", a set of variables 

Author: ```suggestion
confounding variable by estimating a ``substitute confounder'': a set of variables 
```

that once controlled for, renders all variation in the treatment variables of 
interest exogenous. The process of applying the method is actually quite 
straightforward and simple. It proceeds as follows:
\begin{itemize}
	\item First, estimate the substitute confounder using any good latent variable 
	model the modeler chooses. The authors suggest estimating a factor model 
	with k factors on the set of covariates the modeler is interested in. 
	\item Second, check the factor model's accuracy using posterior predictive 
	checks (add details). 
	\item Once a sufficiently accurate latent variable model is recovered, use it 
	to estimate an expected value of the latent variable for each observation, 
	and control for this value in the outcome model, alongside the treatment 
	variables and other covariates of interest. 
\end{itemize}

One main assumption of the deconfounder algorithm is that the data at hand 

> should have multi-cause confounders, thus the title of the paper, "The 

Author: ```suggestion
should only have multi-cause confounders, thus the title of the paper, "The 
```

> blessings of multiple causes". In other words, this method works when the 

Author: Your sentence is totally correct, but I think it's misleading because it doesn't highlight the constraining role of single-cause confounders.
```suggestion
blessings of multiple causes". In other words, this method works when all 
```

unobserved confounders affect multiple of the observed causes (or treatment 
variables) of interest, alongside the outcome. This assumption is weaker than 
the one required for ignorability to hold, which requires the absence of both 
single cause and multi-cause confounders for accurate causal inferences. 

\subsection{Case study: Simulation}
\label{sec:deconfounder-simulation}

The purpose of this section is to investigate the effectiveness of the 
deconfounder algorithm \citep{wang_2019_blessings} in adjusting for unobserved 

> confounding. We use a simulated mode choice data where travel distance 

Author: ```suggestion
confounding. We use simulated mode choice data where travel distance 
```

linearly confounds both travel time and travel cost. We then mask the travel 
distance data and treat it as an unobserved variable. 

We estimate three models:

\begin{itemize}
	\item Model 1: A multinomial logit with the correct original specification, 
	EXCEPT we ommit the travel distance variable in the specification without 
	trying to adjust for it. This model represents the worst case scenario 
	where a modeler ignores, or is unaware of, unobserved confounding.
	\item Model 2: We use the deconfounder algorithm to try to recover the 
	confounder (travel distance). In this method, we use all the variables in 
	each mode's utility to recover that mode's confounder. This is in line 

> 	with the approach taken in \citet{wang_2019_blessings}, where they use all the set 

Author: ```suggestion
	with the approach taken in \citet{wang_2019_blessings}, where they use all the 
```

	of observed variables in the factor model to recover a substitute 
	confounder.
	\item Model 3: We use the deconfounder algorithm to try to recover the 
	confounder (travel distance), but this time, we only use travel time and 
	cost in the factor model, instead of all the variables in the utility 
	specification of each mode. By only using what we know are confounded 
	variables to recover the substitute confounder, our goal is to analyze 

> 	whether we can improve the accuracy of this approach by adopting a more 

Author: Since including extra variables won't bias your factor analysis inferences (e.g. asymptotically, you'll just recover factor loading of zero from irrelevant variables), it is just reducing variance (less params to estimate). Lower variance estimator = higher efficiency estimator (less data needed to get good estimates).
```suggestion
	whether we can improve the efficiency of this approach by adopting a more 
```

> 	whether we can improve the accuracy of this approach by adopting a more 
> 	thoughtful approach to hypothesizing on the confounder variables. This can 

Author: Since including extra variables won't bias your factor analysis inferences (e.g. asymptotically, you'll just recover factor loading of zero from irrelevant variables), it is just reducing variance (less params to estimate). Lower variance estimator = higher efficiency estimator (less data needed to get good estimates).

Also, note that restricting the variables used in the factor analysis is choosing a (strong) degenerate prior distribution for those factor loadings, centered on zero.
```suggestion
	whether we can improve the efficiency of this approach by adopting a stronger 
	prior on which variables are confounded. This can 
```

	be in the form of building and testing candidate causal graphs that 
	try to illustrate this confounding. 
\end{itemize}

> We compare both the estimates of the coefficients on travel time and cost from 
> each of those three models to the true estimates used in the simulation, as well as the distribution of the recovered substitute confounder under each of 
> models 2 and 3 to the true confounder. The main findings of this exercise are the following:

Author: ```suggestion
We compare both the coefficient estimates of travel time and cost from 
each of those three models to the true estimates used in the simulation. We also compare the distribution of the recovered substitute confounder under each of 
models 2 and 3 to the true confounder. The main findings of this exercise are the following:
```

Using the true variables believed to be confounded (i.e. method 3 where only 
travel time and cost are used to recover the confounder) leads to a better 
recovery of the true confounder. Figure \ref{fig:qq-plot-method-2} and \ref{fig:qq-plot-method-3} show a QQ plot of the true 
and recovered confounders under models 2 and 3 respectively. Looking at those 

> figures, one can see that the distribution of the recovered substitute 

Author: ```suggestion
figures, we see that the distribution of the recovered substitute 
```

observed covariates. Please refer to Section \ref{sec:graph-construction} for how to build plausible causal graphs. 

\begin{figure}
   \centering
   \includegraphics[width=0.5\textwidth]{qq-plot-method-2.png}
   \caption{QQ plot of true confounder against recovered confounder using all covariates}
   \label{fig:qq-plot-method-2}
\end{figure}

\begin{figure}
   \centering
   \includegraphics[width=0.5\textwidth]{qq-plot-method-3.png}
   \caption{QQ plot of true confounder against recovered confounder using all covariates}
   \label{fig:qq-plot-method-3}
\end{figure}

% \begin{figure}
% \centering
% \begin{minipage}{.5\textwidth}
%   \centering
%   \includegraphics[width=.4\linewidth]{qq-plot-method-2.png}
%   \captionof{figure}{QQ plot of true confounder against recovered confounder using all covariates}
%   \label{fig:qq-plot-method-2}
% \end{minipage}%
% \begin{minipage}{.5\textwidth}
%   \centering
%   \includegraphics[width=.4\linewidth]{qq-plot-method-3.png}
%   \captionof{figure}{QQ plot of true confounder against recovered confounder using only confounded covariates}
%   \label{fig:qq-plot-method-3}
% \end{minipage}
% \end{figure}

> Additionally, and perhaps most importantly, the effectiveness of the 

Author: Man, this paragraph and the subsection on the simulation study is super clear and a real pleasure to read!

deconfounder algorithm is very sensitive to small errors and misfits in the 
recovered confounder. Although method 3 returns a relatively good fit of the 
true confounder (based on the QQ plot), the adjusted coefficients on travel 
time and cost do not exhibit any reduction in the bias resulting from 

> ommitting the true confounder, and the coefficients on the recovered 

Author: Splitting up a long sentence so each new sentence expresses a single (simpler) thought.
```suggestion
omitting the true confounder. Moreover, the coefficients on the recovered 
```

> confounder are highly insignificant. This raises questions about the 
> usefulness of the deconfounder algorithm in practice. While this limitation 
> may raise questions on the usefulness of the deconfounder algorithm, it is 

Author: Removing sentence duplication
```suggestion
confounder are highly insignificant. This raises questions about the 
usefulness of the deconfounder algorithm in practice. Limitations notwithstanding, it is 
```

> important to point out that it is not only a by-product of the deconfounder 

Author: ```suggestion
important to point out that sensitivity to small errors and misfits is not only a by-product of the deconfounder 
```

algorithm itself. In fact, it is one of the built in characteristic of ommitted

> variable bias, and perhaps more broadly, the problem of error-in-variables in 

Author: Interesting. I think of error-in-variables as a special case of omitted variable bias, not the other way around. For example, there are types of omitted variable bias that are not errors-in-variables. Latent confounding is great example. I don't know that there are types of errors-in-variables that are not omitted variables though.
